mlir/lib/Quantum/Transforms/AdjointPatterns.cpp

```diff
@@ -156,6 +156,10 @@ class AdjointGenerator {
              llvm::zip(gate.getQubitResults(), gate.getQubitOperands())) {
             remappedValues.map(qubitOperand, remappedValues.lookup(qubitResult));
         }
+        for (const auto &[qubitResult, qubitOperand] :
+             llvm::zip(gate.getCtrlQubitResults(), gate.getCtrlQubitOperands())) {
+            remappedValues.map(qubitOperand, remappedValues.lookup(qubitResult));
+        }
```

Comment:
I'm going to propose an update to the QuantumGate interface so that this is not required at all.